finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -104,6 +139,11 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
     transport.read().rescue(wrapRemoteEx)
 
   protected[this] def demux(): Future[Unit] = {
+    detector.onClose.ensure {
```

Comment:
This will register an additional callback every time this dispatcher receives a frame.  I think this should probably be set up when the detector is created.